APCS-秘密差

ChunYu-Laboratory opened the pull request.

ChunYu-Laboratory committed:
Create 秘密差.py

ChunYu-Laboratory committed:
Create Problem.pdf

bruce3557 commented on APCS/秘密差/秘密差.py:
```diff
@@ -0,0 +1,33 @@
+A = 0
+B = 0
+for i in a :
+    A += i
```
可以改成
`A = sum(a)`

bruce3557 commented on APCS/秘密差/秘密差.py:
```diff
@@ -0,0 +1,33 @@
+    b = num[::2]
+else :
+    a = num[::2]
+    b = num[1::2]
```
提供另一個想法xD
我會直接loop一次num, 有點像這樣
```python
sign = 1
total = 0
for digit in num:
  total += sign * digit
  sign = -sign
answer = abs(total)
```

ChunYu-Laboratory committed:
Update

ChunYu-Laboratory committed:
Update 秘密差.py

Apply the sum function instead of the original calculation loop in section 5

ChunYu-Laboratory committed:
Update 秘密差.py

Change the core algorithm
Use the loop to sweep through the num list to perform sign reversal operation on the adjacent numbers and add the total to replace the original odevity number of digits are stored in different lists to be calculated in section 4
